src/panels/config/automation/blueprint-automation-editor.ts

```diff
@@ -129,27 +128,19 @@ export class HaBlueprintAutomationEditor extends LitElement {
                           breaks
                           .content=${value?.description}
                         ></ha-markdown>
-                        ${value?.selector
-                          ? html`<ha-selector
-                              .hass=${this.hass}
-                              .selector=${value.selector}
-                              .key=${key}
-                              .value=${(this.config.use_blueprint.input &&
-                                this.config.use_blueprint.input[key]) ??
-                              value?.default}
-                              .disabled=${this.disabled}
-                              .required=${value?.default === undefined}
-                              @value-changed=${this._inputChanged}
-                            ></ha-selector>`
-                          : html`<ha-textfield
-                              .key=${key}
-                              .required=${value?.default === undefined}
-                              .value=${(this.config.use_blueprint.input &&
-                                this.config.use_blueprint.input[key]) ??
-                              value?.default}
-                              .disabled=${this.disabled}
-                              @input=${this._inputChanged}
-                            ></ha-textfield>`}
+                        ${html`<ha-selector
+                          .hass=${this.hass}
+                          .selector=${value?.selector ?? { text: undefined }}
+                          .key=${key}
+                          .disabled=${this.disabled}
+                          .required=${value?.default === undefined}
+                          .placeholder=${value?.default}
+                          .value=${this.config.use_blueprint.input &&
+                          key in this.config.use_blueprint.input
+                            ? this.config.use_blueprint.input[key]
+                            : value?.default}
+                          @value-changed=${this._inputChanged}
+                        ></ha-selector>`}
```

Comment:
Use a text selector by default, instead of an ha-textfield, when no selector is defined. That keeps things a little bit simpler, since the behavior of text selector and ha-textfield were subtly different, and this is less cases to have to be concerned with.